content/docs/setup/kubernetes/mesh-expansion/index.md

```diff
@@ -5,272 +5,297 @@ weight: 50
+* You have already set up Istio on Kubernetes. If you haven't done so, you can find out how in the [Installation guide](/docs/setup/kubernetes/quick-start/).
 
-* The machine must have IP connectivity to the endpoints in the mesh. This
+* Mesh expansion machines must have IP connectivity to the endpoints in the mesh. This
 typically requires a VPC or a VPN, as well as a container network that
 provides direct (without NAT or firewall deny) routing to the endpoints. The machine
 is not required to have access to the cluster IP addresses assigned by Kubernetes.
 
-* The Istio control plane services (Pilot, Mixer, Citadel) and Kubernetes DNS server must be accessible
-from the VMs. This is typically done using an [Internal Load
-Balancer](https://kubernetes.io/docs/concepts/services-networking/service/#internal-load-balancer).
-You can also use NodePort, run Istio components on VMs, or use custom network configurations,
-separate documents will cover these advanced configurations.
+* Mesh expansion VMs must have access to a DNS server that resolves names to cluster IP addresses. Options
+include exposing the Kubernetes DNS server through an internal load balancer, using a Core DNS
+server, or configuring the IPs in any other DNS server accessible from the VM.
+
+* If you haven't already enabled mesh expansion [at install time with Helm](/docs/setup/kubernetes/helm-install/), you have
+installed the [Helm client](https://docs.helm.sh/using_helm/). You'll need it to enable mesh expansion for the cluster.
 
 ## Installation steps
 
 Setup consists of preparing the mesh for expansion and installing and configuring each VM.
 
-An example script to help with Kubernetes setup is available as part of the release bundle called
-[`install/tools/setupMeshEx.sh`]({{< github_file >}}/install/tools/setupMeshEx.sh). Check the script content and environment variables supported (like GCP_OPTS).
+### Preparing the Kubernetes cluster for expansion
 
-An example script to help configure a machine is available as part of the release bundle called [`install/tools/setupIstioVM.sh`]({{< github_file >}}/install/tools/setupIstioVM.sh).
-You should customize it based on your provisioning tools and DNS requirements.
+The first step when adding non-Kubernetes services to an Istio mesh is to configure the Istio installation itself, and
+generate the configuration files that let mesh expansion VMs connect to the mesh. To prepare the
+cluster for mesh expansion, run the following commands on a machine with cluster admin privileges:
 
-### Preparing the Kubernetes cluster for expansion
+1.  Ensure that mesh expansion is enabled for the cluster. If you did not specify `--set global.meshExpansion=true` at
+    install time with Helm, there are two options for enabling mesh expansion, depending on how you originally installed
+    Istio on the cluster: 
+    
+    *   If you installed Istio with Helm and Tiller, run `helm upgrade` with the new option:
+    
+    {{< text bash >}}
```

Comment:
some of this document has a CR after the text bash and before the terminating md block.

There is no need for a CR between the text block and the beginning of the text.

This is an example of the work done correctly.

Reply:
done!